src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,45 @@
 package core.basesyntax;
 
-class Pair {
+public class Pair<F, S> {
 
+    private F first;
+    private S second;
+
+    private Pair(F first, S second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public static <F, S> Pair<F, S> of(F first, S second) {
+        return new Pair<>(first, second);
+    }
+
+    public F getFirst() {
+        return first;
+    }
+
+    public S getSecond() {
+        return second;
+    }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (object == null || getClass() != object.getClass()) {
+            return false;
+        }
+        Pair<?, ?> pair = (Pair<?, ?>) object;
```

Comment:
why are we using wildcard here? you know which type you work with

Reply:
about this we were talking on the first Q&A sessinon, moreover compiler pointed on this with yellow marked. after using wildcard the marker dissapeared. anywhay I can remove it or insert type

Reply:
let's use types here and ignore this message from IDE
this is our method implementation, and we can ensure ourselves that this cast is secure

Reply:
done